lib/config-win32.h

```diff
@@ -332,7 +332,7 @@
 #define SIZEOF_CURL_OFF_T 8
 
 /* Define to the size of `off_t', as computed by sizeof. */
-#ifndef SIZEOF_OFF_T
+#if defined(__MINGW32__)
```

Comment:
Besides MSVC, MinGW and MinGW-w64, what other compilers could use config-win32? If we know the size the compilers will use I think we should be explicit about it. For example

```suggestion
#if defined(_MSC_VER)
#define SIZEOF_OFF_T 4
#elif defined(__MINGW32__)
```

Reply:
Also note original mingw uses 32-bit off_t, so this would have to be further revised to only account when mingw-w64 is using a 64-bit off_t which may be all? some? of the time

Reply:
@jay: Thanks for the old-mingw info, I keep forgetting about it. As for the rest, do you think it's best to explicitly list all platforms that need 4?

Like so:
```c
#if defined(_MSC_VER)
#define SIZEOF_OFF_T 4
#elif defined(__MINGW64_VERSION_MAJOR)
#define SIZEOF_OFF_T 8
#elif defined(__MINGW32__)
#define SIZEOF_OFF_T 4
#endif
```

Or we could stay short, like:
```c
#if defined(__MINGW64_VERSION_MAJOR)
#define SIZEOF_OFF_T 8
#else
#define SIZEOF_OFF_T 4
#endif
```

Reply:
If you are sure that MinGW-w64 uses off_t 64 bits. MS data type for off_t is 32 bits.

Reply:
I verified, and it's 64 bits when `_FILE_OFFSET_BITS == 64` (otherwise it's 32). This is so for mingw-w64 1.0 and later releases (= all releases), thanks to this [commit](https://github.com/Alexpux/mingw-w64/commit/4d3b28a9929ea58511e7165cb7eb1bcdd01151ad) (from 2011-02-19).

This should cover everything we know so far:
```c
#if defined(__MINGW64_VERSION_MAJOR) && \
  defined(_FILE_OFFSET_BITS) && (_FILE_OFFSET_BITS == 64)
#define SIZEOF_OFF_T 8
#else
#define SIZEOF_OFF_T 4
#endif
```
and moving the block under the defaulting logic for `_FILE_OFFSET_BITS`.